azure/services/scalesets/scalesets_test.go

```diff
@@ -214,6 +214,18 @@ func TestReconcileVMSS(t *testing.T) {
 				s.ScaleSetSpec(gomockinternal.AContext()).Return(&spec).AnyTimes()
 			},
 		},
+		{
+			name:          "validate spec success: Memory is float",
+			expectedError: "",
+			expect: func(g *WithT, s *mock_scalesets.MockScaleSetScopeMockRecorder, r *mock_async.MockReconcilerMockRecorder, m *mock_scalesets.MockClientMockRecorder) {
+				s.DefaultedAzureServiceReconcileTimeout().Return(reconciler.DefaultAzureServiceReconcileTimeout)
+				spec := newDefaultVMSSSpec()
+				spec.Size = "VM_SIZE_MEM_FLOAT"
+				spec.Capacity = 2
+				spec.SSHKeyData = sshKeyData
+				s.ScaleSetSpec(gomockinternal.AContext()).Return(&spec).AnyTimes()
+			},
+		},
```

Comment:
Looks like this test case is causing an error. I would personally test the new logic by adding a float size to `newDefaultVMSSSpec`. That way, the first few cases in `TestReconcileVMSS` will cover it.

Reply:
Thanks for taking a look. I think I got the test right this time.